R/between.R

```diff
@@ -16,15 +16,14 @@ between = function(x, lower, upper, incbounds=TRUE, NAbounds=TRUE, check=FALSE)
     stopifnot(is.px(x), is.px(lower), is.px(upper)) # nocov # internal
   }
   # POSIX check timezone match
-  if (is.px(x) && is.px(lower) && is.px(upper)) {
-    tzs = sapply(list(x,lower,upper), function(x) {
+  if (!ignore_tzone && is.px(x) && is.px(lower) && is.px(upper)) {
+    tzs = vapply_1c(list(x,lower,upper), function(x) {
       attr(x, "tzone", exact=TRUE) %||% ""
     })
     # lower/upper should be more tightly linked than x/lower, so error
     #   if the former don't match but only inform if they latter don't
     if (tzs[2L]!=tzs[3L]) {
       stopf("'between' lower= and upper= are both POSIXct but have different tzone attributes: %s. Please align their time zones.", brackify(tzs[2:3], quote=TRUE))
```

Comment:
I'm OK changing this to a warning if you'd like -- please just add a number of stress tests that the code WAI when time zones don't match.

Reply:
Hi @r2evans WDYT, do you want to demote this to `warning()`? Or perhaps leave as a separate PR?